src/main/java/org/opensearch/jobscheduler/model/JobDetails.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.jobscheduler.model;
+
+import org.opensearch.common.xcontent.ToXContent;
+import org.opensearch.common.xcontent.ToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+
+/**
+ * This model class stores the job details of the extension.
+ */
+public class JobDetails implements ToXContentObject {
+
+    /**
+     * jobIndex from the extension.
+     */
+    private String jobIndex;
+
+    /**
+     * jobType from the extension.
+     */
+    private String jobType;
+
+    /**
+     * jobParser action to trigger the response back to the extension.
+     */
+    private String jobParameterAction;
+
+    /**
+     * jobRunner action to trigger the response back to the extension.
+     */
+    private String jobRunnerAction;
+
+    public static final String JOB_INDEX = "job_index";
+    public static final String JOB_TYPE = "job_type";
+    public static final String JOB_PARAMETER_ACTION = "job_parser_action";
+    public static final String JOB_RUNNER_ACTION = "job_runner_action";
+
+    public JobDetails() {}
+
+    public JobDetails(String jobIndex, String jobType, String jobParameterAction, String jobRunnerAction) {
+        this.jobIndex = jobIndex;
+        this.jobType = jobType;
+        this.jobParameterAction = jobParameterAction;
+        this.jobRunnerAction = jobRunnerAction;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
+        XContentBuilder xContentBuilder = builder.startObject();
+        if (jobIndex != null) {
+            xContentBuilder.field(JOB_INDEX, jobIndex);
+        }
+        if (jobType != null) {
+            xContentBuilder.field(JOB_TYPE, jobType);
+        }
+        if (jobParameterAction != null) {
+            xContentBuilder.field(JOB_PARAMETER_ACTION, jobParameterAction);
+        }
+        if (jobRunnerAction != null) {
+            xContentBuilder.field(JOB_RUNNER_ACTION, jobRunnerAction);
+        }
+        return xContentBuilder.endObject();
```

Comment:
Nulls are dangerous and usually best avoided.  Personal preference is to try to use `Optional` instead.  I can be convinced to stick with null but will ask for it to be amply documented in that case.

Looking at all the classes in this PR, it seems that we always require `jobType` to be non-null, but the other values are allowed to be null.  That isn't obvious from looking at this class in isolation.  I suggest:
1. Test the value of `jobType` before returning here, and throw an appropriate parsing exception here if it is null.
2. Add the `@Nullable` annotation to the getters for the other three fields.
3. Add the `@Nullable` annotation to each of the (allowed to be null) parameters on the constructor.
4. Add a javadoc somewhere (perhaps the class itself, or this parsing method) indicating the allowed/disallowed nulls.

Reply:
done